proxy_manager.go

```diff
@@ -231,3 +232,31 @@ func (m *proxyManager) getFlakeIDGeneratorConfig(name string) FlakeIDGeneratorCo
 func makeProxyName(serviceName string, objectName string) string {
 	return fmt.Sprintf("%s%s", serviceName, objectName)
 }
+
+type proxyDestroyer interface {
+	Destroy(ctx context.Context) error
+}
+
+func (m *proxyManager) destroyProxies(ctx context.Context) {
+	for key, p := range m.proxies {
+		if p == nil {
+			m.serviceBundle.Logger.Log(logger.WeightError, func() string {
+				return fmt.Sprintf("proxy %s key cannot be destroyed, given proxy argument is nil", key)
+			})
+			continue
+		}
```

Comment:
Is it feasible that `p == nil`? I think we should remove this check.

Reply:
Is there a reason to set an entry to `nil`? I think that can only happen by mistake.

Reply:
Removed related section, [694aee3](https://github.com/hazelcast/hazelcast-go-client/pull/772/commits/694aee3b72c6dd7465c5d3bde999c2fdc292f28e).